Assets/Scripts/ExplotionController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.Mathematics;
+using Unity.VisualScripting;
+using UnityEngine;
+using UnityEngine.VFX;
+
+public class ExplotionController : MonoBehaviour
+{
+    [SerializeField] 
+    public int damage;
+
+    [SerializeField]
+    private VisualEffect visualEffect;
+
+    [SerializeField]
+    public float radius;
+
+    [SerializeField]
+    private LayerMask hitBoxLayers;
+    
+    private HashSet<HealthController> hitHealthControllers = new HashSet<HealthController>();
+    
+
+    private void Start()
+    {
+        visualEffect.enabled = false;
+    }
+
+    public void OnDrawGizmos()
+    {
+        Gizmos.DrawWireSphere(transform.position,radius);
+    }
+
+    public void Explode()
+    {
+        visualEffect.enabled = true;
+        visualEffect.SendEvent("OnPlay");
+        Collider[] colliderList = Physics.OverlapSphere(transform.position, radius, hitBoxLayers);
+        
+        foreach (Collider collider in colliderList)
+        {
+            HitOrMiss(collider);
+        }
+    }
+
+    private void HitOrMiss(Collider collider)
+    {
+        HitboxController controller = collider.GetComponent<HitboxController>();
+        if (!controller.health || !hitHealthControllers.Contains(controller.health))
+        {
+            hitHealthControllers.Add(controller.health);
+            float distdam = math.pow((Vector3.Distance(collider.transform.position, transform.position) / radius), 2);
+            float damageFinal = (1-distdam)*damage;
+            Debug.Log(damageFinal+": Dette er skaden");
+            Debug.Log("Avstanden er "+ Vector3.Distance(collider.transform.position, transform.position)/radius);
+            Debug.Log("Posisjonene er :"+transform.position+" for Barrel || "+collider.transform.position+" for hitbox");
```

Comment:
Please remove superfluous logs from your PR, we're able to test this through the damage to the players now.